exercises/dot-dsl/tests/dot-dsl.rs

```diff
@@ -0,0 +1,128 @@
+extern crate dot_dsl;
+
+use dot_dsl::graph::graph_items::edge::Edge;
+use dot_dsl::graph::graph_items::node::Node;
+use dot_dsl::graph::Graph;
+
+use std::collections::HashMap;
+
+#[test]
+fn test_empty_graph() {
+    let graph = Graph::new();
+
+    assert!(graph.nodes.is_empty());
+
+    assert!(graph.edges.is_empty());
+
+    assert!(graph.attrs.is_empty());
+}
+
+#[test]
+#[ignore]
+fn test_graph_with_one_node() {
+    let nodes = vec![Node::new("a")];
+
+    let graph = Graph::new().with_nodes(&nodes);
+
+    assert!(graph.edges.is_empty());
+
+    assert!(graph.attrs.is_empty());
+
+    assert_eq!(graph.nodes, vec![Node::new("a")]);
+}
+
+#[test]
+#[ignore]
+fn test_graph_with_one_node_with_keywords() {
+    let nodes = vec![Node::new("a").with_attrs(&[("color", "green")])];
+
+    let graph = Graph::new().with_nodes(&nodes);
+
+    assert!(graph.edges.is_empty());
+
+    assert!(graph.attrs.is_empty());
+
+    assert_eq!(
+        graph.nodes,
+        vec![Node::new("a").with_attrs(&[("color", "green")])]
+    );
+}
+
+#[test]
+#[ignore]
+fn test_graph_with_one_edge() {
+    let edges = vec![Edge::new("a", "b")];
+
+    let graph = Graph::new().with_edges(&edges);
+
+    assert!(graph.nodes.is_empty());
+
+    assert!(graph.attrs.is_empty());
+
+    assert_eq!(graph.edges, vec![Edge::new("a", "b")]);
+}
+
+#[test]
+#[ignore]
+fn test_graph_with_one_attribute() {
+    let graph = Graph::new().with_attrs(&[("foo", "1")]);
+
+    let mut expected_attrs = HashMap::new();
```

Comment:
The [`maplit` crate](https://crates.io/crates/maplit) is good for test cases like this.

Reply:
I thought external dependencies were not permitted in the test suite?

Reply:
They're discouraged but not strictly forbidden, and I don't recall a written policy. What it really comes down to is a judgment call on the part of the track moderators. In my judgment, using `maplit` macros increases test clarity, so I allow it.